### DISCUSSION TITLE
Improving the communication with the community in a more transparently way

### DISCUSSION THREAD

**Discussion by Shinigami92** [lodash/lodash#5732] [General]
Hello @jdalton :wave: 

I would like to suggest to improve the communication with the lodash community with some more or less simple changes

1. open the "Discussions" feature on the repository
2. Pinning (even closed) issues like https://github.com/lodash/lodash/issues/5716
3. Writing a statement why the repo radically closed all issues and PRs

---

I just assume and find out that lodash is now working on a full rewrite on v5 driven by Bun support
I support that! lodash is used by MANY packages and nearly every project feels like having a 50% that any sub-dependency contains lodash somehow.
With that responsibility the lodash maintainers should try to not mess with their community.

Maybe I'm biased about such radical changes due to I'm the maintainer of a repo that we took over due to a radical situation... but I just want a healthy community 🤗

**Comment by cwtuan** [comment]
I want to express gratitude for the incredible work you've done in maintaining Lodash over the years. It's no small feat, and Lodash remains an essential utility for countless developers.

I understand that dealing with a large number of GitHub issues can be overwhelming, especially when many of them are discussions rather than actual bug reports. To help maintain a productive community around Lodash, I may offer some suggestions:

**Communication Channels:**
Consider setting up a dedicated discussion forum or a platform like Discord or Slack for users to engage in discussions. This can help keep GitHub issues primarily for bug reports and feature requests.

**Automate Repetitive Tasks:**
Automate tasks like responding to common questions using GitHub Actions or bots.

**Moderation and Labeling:**
Appoint moderators or a team of trusted contributors who can help triage and label issues appropriately.

**Issue Templates:**
Implement issue templates for bug reports, feature requests, and discussions. These templates can guide users in providing the necessary information and context when creating an issue.

By the way, I'd like to point out that Lodash is still recognized as the number one JavaScript utility library according to [lodash vs underscore vs ramda](https://npm-compare.com/lodash,underscore,ramda). This reflects its importance within the JavaScript ecosystem. Thank you for your hard work to the Lodash project.

**Comment by ashishpandey001** [comment]
As @Shinigami92 and @cwtuan said.

lodash has been the missing utilities lib that I use in all of my projects.

Much love and appreciation for all the hard work and thankless contributions.

Hoping to see v5 and would be glad to contribute into having discussions around optimisations for various runtime flavours.

Keep up the good work everyone!

**Comment by jdalton** [comment]
@Shinigami92 

Improving engagement is a great goal. 
I haven't pinned issues much so thank you for highlighting that option.
Declaring issue bankruptcy is a recognition that after ~4 years of recovering from burn out there's no way to acknowledge the 600+ issues/suggestions/prs/questions that have built-up in that time. 

Realistically I'll be 
 * dusting off the tech debt.
 * make adjustments to modernize and reduce the scope
 * then look at adding features
 
 Lodash 4 will get surgical, handcrafted updates that will aim to preserve existing compat. A lot of benefit can be had for making non-breaking tweaks there. Lodash 5 (main branch) will be where the newer work occurs.

**Comment by jdalton** [comment]
Thanks for the kind words y'all ❤️

**Comment by jdalton** [comment]
I've turned on the "discussion" feature ✅

**Comment by Shinigami92** [comment]
> Declaring issue bankruptcy is a recognition that after ~4 years of recovering from burn out there's no way to acknowledge the 600+ issues/suggestions/prs/questions that have built-up in that time.

I didn't know that and I feel sorry for you that you ran into a burnout
Something I also definitely don't want to run into myself

But let me tell you: try to built up a community and find people that help you in the project
There are even potential people that wants to make community-management

A Discord-Server is a first good start (=> https://github.com/lodash/lodash/discussions/5722)

- https://github.com/lodash/lodash/discussions/5722

At first it might sound like a lot of work to build up a community, but some day it pays back

I know what I'm talking about, because I'm in huge ecosystems like Vite, Faker, the Antfu-universe and also I participate in other Vue related communities like Quasar and so on

Another idea should be to open up a https://opencollective.com page so the community can pay some money for the great work
This money could not only be used to compensate yourself, but also to pay some people for community management and other stuff

Sadly it looks like someone already claimed https://opencollective.com/lodash so you would need to claim that back or just name it differently

An example of what you could use opencollective:
We once payed $300 to someone because he put a lot of effort into the Color Module of Faker https://opencollective.com/fakerjs/expenses/86030

- https://github.com/lodash/lodash/discussions/5723

---

also here is a doc for how to pin an issue: https://docs.github.com/en/issues/tracking-your-work-with-issues/pinning-an-issue-to-your-repository

**Comment by ygj6** [comment]
Lodash is a widely used front-end library, thanks to the hard work of @jdalton! I'm glad lodash has plans for a new version! In fact, my team is also heavily using lodash, so we're very happy to see that.

**Comment by devstein** [comment]
>Automate Repetitive Tasks:
Automate tasks like responding to common questions using GitHub Actions or bots.

I recommend checking out [Dosu](https://github.com/apps/dosu-beta). Dosu helps open source maintainers encourage contributors, organize issues, triage bugs, answer questions, and more. See it in action:

https://github.com/jerryjliu/llama_index/issues/7666#issuecomment-1718137079
https://github.com/langchain-ai/langchainjs/issues/2528#issuecomment-1708091668
https://github.com/langchain-ai/langchain/discussions/10328#discussioncomment-6941135

**Comment by Shinigami92** [comment]
@jdalton now that GitHub Discussions are open, I suggest that you click in this issue and in #5721 on the "convert to discussion" button/link on the right (in desktop view)

**Comment by wangsongc** [comment]
Hi @jdalton 

Is there any way I can contribute to the release of a new version for lodash?

I've come across the discussion about the upcoming lodash version in https://github.com/lodash/lodash/discussions/5732#discussioncomment-7060254, but it appears that there haven't been substantial changes in the codebase, and there's no clear release date yet (not to rush things—I personally find it an effective way to manage a project). Therefore, I'm reaching out to inquire if there's anything I can do to assist in expediting the release of a new version for lodash. I'm more than willing to offer my support wherever I can help : )
